keychain.h

```diff
@@ -0,0 +1,70 @@
+#include <assert.h>
+
+#include <optional>
+#include <string>
+#include <variant>
+
+namespace keychain {
+
+class Keychain {
+   public:
+    struct Error {
+        std::string message;
+    };
+    using Success = std::monostate;
+    using Password = std::string;
+    using Result = std::variant<Success, Password, Error>;
+
+    static Result getPassword(const std::string &service,
+                              const std::string &user);
+    static Result setPassword(const std::string &service,
+                              const std::string &user,
+                              const std::string &password);
+    static Result deletePassword(const std::string &service,
+                                 const std::string &user);
+
+   public:
+    /*! \brief Construct a Keychain specific to a service/user combination
+     *
+     * \param service Identifier for the service whose password should be stored
+     *                or updated. Note that a platform specific service name
+     *                will be used based on the provided identifier.
+     *                \sa getServiceName
+     * \param user
+     */
+    Keychain(const std::string &service, const std::string &user);
+
+    std::optional<std::string> getPassword();
```

Comment:
I think, if we already keep state in the `Keychain` class, we should only keep the `service` and allow to manage multiple passwords with one instance. Otherwise I don’t really see the benefit of a strateful object....

Reply:
I see. Maybe I'm not quite sure how we want to use this. As of now, it allows you to manage exactly one password. You'd make another instance to manage another password.

Assuming our application always identifies as the same service, it makes sense to fix the service. I'm not sure if our application would want to store passwords for more than one user with this object.

In the case we discussed, I'm actually wondering how we'd want to use this. Assume we want to store two different generic tokens from our application. What `service` / `user` would each of the tokens be stored as?

Reply:
The  outcome of this discussion is this comment: https://github.com/hrantzsch/keychain/pull/2/files#r346188920